src/libraries/Microsoft.Extensions.Options/ref/Microsoft.Extensions.Options.cs

```diff
@@ -163,37 +163,37 @@ public partial class OptionsBuilder<TOptions> where TOptions : class
         public virtual Microsoft.Extensions.Options.OptionsBuilder<TOptions> PostConfigure<TDep1, TDep2, TDep3, TDep4, TDep5>(System.Action<TOptions, TDep1, TDep2, TDep3, TDep4, TDep5> configureOptions) where TDep1 : class where TDep2 : class where TDep3 : class where TDep4 : class where TDep5 : class { throw null; }
         public virtual Microsoft.Extensions.Options.OptionsBuilder<TOptions> Validate(System.Func<TOptions, bool> validation) { throw null; }
         public virtual Microsoft.Extensions.Options.OptionsBuilder<TOptions> Validate(System.Func<TOptions, bool> validation, string failureMessage) { throw null; }
-        public virtual Microsoft.Extensions.Options.OptionsBuilder<TOptions> Validate<TDep>(System.Func<TOptions, TDep, bool> validation) { throw null; }
```

Comment:
Why are these changes necessary?

Reply:
`TDep` is used in `sp.GetRequiredService<TDep>()`, and it has a `notnull` constraint, so `TDep` should also have it